src/routes/login.ts

```diff
@@ -57,12 +57,21 @@ export const createLoginRoute: RouteCreator =
     // It is probably a bit strange to have a logout URL here, however this screen
     // is also used for 2FA flows. If something goes wrong there, we probably want
     // to give the user the option to sign out!
-    const logoutUrl =
-      (
-        await frontend
-          .createBrowserLogoutFlow({ cookie: req.header("cookie") })
-          .catch(() => ({ data: { logout_url: "" } }))
-      ).data.logout_url || ""
+    const getLogoutUrl = async (loginFlow: LoginFlow) => {
+      return (
+        (
+          await frontend
+            .createBrowserLogoutFlow({
+              cookie: req.header("cookie"),
+              returnTo:
+                (return_to && return_to.toString()) ||
+                loginFlow.return_to ||
+                "",
+            } as { cookie: string; returnTo: string })
+            .catch(() => ({ data: { logout_url: "" } }))
```

Comment:
Why not try { .. } catch () { }?

Reply:
can do, but does it change anything? :p

Reply:
You don't mix .then/.catch with await :) it's a codestyle nit...